app/models/pipeline_run.rb

```diff
@@ -596,24 +595,25 @@ def load_taxons(downloaded_json_path, refined = false)
     return if loaded_records > 0
 
     # only keep counts at certain taxonomic levels
-    taxon_counts_attributes_filtered = []
     acceptable_tax_levels = [TaxonCount::TAX_LEVEL_SPECIES]
     acceptable_tax_levels << TaxonCount::TAX_LEVEL_GENUS if multihit?
     acceptable_tax_levels << TaxonCount::TAX_LEVEL_FAMILY if multihit?
-    pipeline_output_dict['taxon_counts_attributes'].each do |tcnt|
+    taxon_counts_attributes_filtered = taxon_counts_attributes.select do |tcnt|
       # TODO:  Better family support.
-      if acceptable_tax_levels.include?(tcnt['tax_level'].to_i) && !invalid_family_call?(tcnt)
-        taxon_counts_attributes_filtered << tcnt
-      end
+      acceptable_tax_levels.include?(tcnt['tax_level'].to_i) && !invalid_family_call?(tcnt)
     end
     # Set created_at and updated_at
     current_time = Time.now.utc # to match TaxonLineage date range comparison
+    tcnt_attrs_to_merge = {
+      'created_at' => current_time,
+      'updated_at' => current_time,
+      'pipeline_run_id' => id
+    }
     taxon_counts_attributes_filtered.each do |tcnt|
-      tcnt["created_at"] = current_time
-      tcnt["updated_at"] = current_time
       tcnt["count_type"] += "+" if refined
+      tcnt.merge!(tcnt_attrs_to_merge)
     end
-    update(taxon_counts_attributes: taxon_counts_attributes_filtered)
+    TaxonCount.import taxon_counts_attributes_filtered
```

Comment:
nit: keep pattern using parentheses?

Reply:
Is the time for this operation logged automatically somewhere?
Could be interesting how it evolves and behaves for different samples.

Reply:
Unfortunately this module doesn't have good logs, and this is why it is so hard to find the root cause of these deadlocks. That’s the reason I suggested to create a log strategy for this module, ideally we should have START/END/ERROR log events for each job executed and send it all to Datadog.

I measured this specific command and, as expected, it is extremely faster (30x), since it inserts all the lines using a single INSERT ... VALUES values_list, drastically reducing the network traffic. Also it helps MySQL to create a better lock strategy and transaction log management, since it knows upfront all rows that are being inserted and can check eventual PK violations at once (this is the equivalent for a SELECT ... IN (...), versus multiple SELECT statements).

You can see in this [Google Slides](https://docs.google.com/presentation/d/1iKHUAGGNLKOMAVPJwfoDCdvzpqgVdS4JTMcxjYkNa00/edit?usp=sharing), page 9, some metrics that I gathered in my local environment .

Reply:
> nit: keep pattern using parentheses?
I added it in my new commit to satisfy style preferences from our team.

I feel like it is really important we have an automatic linter ruler being executed on every commit. If we have that, it would make clear what are the code style preferences of our team, and also it would save our time during reviews from doing a task that can (and should) be automated.